### PR TITLE
Add regression test

### PR DESCRIPTION
**Pull request by RikkiGibson**
Adds regression test for #46461

### CODE REVIEW

**Review by AlekseyTs — APPROVED** [review]
LGTM (iteration 1)

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -113,6 +113,40 @@ public void M<T>(bool condition, T t)
...
+
+            comp.VerifyDiagnostics(
+                // (10,37): warning CS8603: Possible null reference return.
+                //         M1<bool, string>(true, b => b ? "" : default); // 1
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, @"b ? """" : default", isSuppressed: false).WithLocation(10, 37),
+                // (14,37): warning CS8603: Possible null reference return.
+                //         M1<bool, string>(true, b => b ? "" : null); // 2
+                Diagnostic(ErrorCode.WRN_NullReferenceReturn, @"b ? """" : null", isSuppressed: false).WithLocation(14, 37));
+        }
+
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs:149]: Could we include the verbatim test from the bug report, or something very close to it?

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs:149]: I felt reluctant to do this due to the complexity and potential dependence on framework nullable annotations.

**Pull request merged**